### PR TITLE
feat: define constants for common HTTP status codes

### LINKED ISSUE #141: Consider adding constants for common HTTP status codes
We have not defined any of the typical constants for HTTP status codes, which means we are either hardcoding the literals (`204`, `503`, etc.) or defining the constants locally to make it easier to read the intent (e.g. `auto constexpr kNoContent = 204;`).

Maybe we should define the most common constants as inline static variables in `HttpRequest`.

The list does not need to be exhaustive to be useful, we can start with the codes from here (minus all the WebDAV stuff):

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

### PR DESCRIPTION
**Pull request by coryan**
Fixes #141

### CODE REVIEW

File: google/cloud/functions/http_response.h [hunk 1]
```diff
@@ -41,6 +41,80 @@ class HttpResponse {
...
+  inline static auto constexpr kMultipleChoice = 300;
+  inline static auto constexpr kMovedPermanently = 301;
+  inline static auto constexpr kFound = 302;
+  inline static auto constexpr kSeeOther = 303;
+  inline static auto constexpr kNotModified = 304;
+  inline static auto constexpr kUseProxy = 305;
+  // inline static auto constexpr kReserved = 306;
+  inline static auto constexpr kTemporaryRedirect = 307;
+  inline static auto constexpr kPermanentRedirect = 308;
+  // Start of Client Error Responses (4XX)
```
Comment (devbww) [inline google/cloud/functions/http_response.h:74]: Nit: "Responses" was in lowercase above.

File: google/cloud/functions/http_response.h [hunk 2]
```diff
@@ -41,6 +41,80 @@ class HttpResponse {
...
+  inline static auto constexpr kUnprocessableEntity = 422;
+  inline static auto constexpr kLocked = 423;
+  inline static auto constexpr kFailedDependency = 424;
+  inline static auto constexpr kTooEarly = 425;
+  inline static auto constexpr kUpgradeRequired = 426;
+  inline static auto constexpr kPreconditionRequired = 428;
+  inline static auto constexpr kTooManyRequests = 429;
+  inline static auto constexpr kRequestHeaderFieldsTooLarge = 431;
+  inline static auto constexpr kUnavailableForLegalReasons = 451;
+  // Start of Server Error Responses (5XX)
```
Comment (devbww) [inline google/cloud/functions/http_response.h:104]: Ditto.

**Review by devbww — APPROVED** [review]

File: google/cloud/functions/http_response.h (hunk 1 above)
Comment (coryan) [reply google/cloud/functions/http_response.h]: Fixed. Also removed the `Start of ` seemed redundant on a second look.

File: google/cloud/functions/http_response.h (hunk 2 above)
Comment (coryan) [reply google/cloud/functions/http_response.h]: Fixed.

**Review by devbww — APPROVED** [review]

**Pull request merged**